src/Form/PolicyForm.php

```diff
@@ -0,0 +1,224 @@
+      '#default_value' => $current_role,
+      '#required' => TRUE,
+      '#disabled' => !$entity->isNew(),
+    ];
+
+    $form['minimumRequiredConstraints'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Minimum required constraints'),
+      '#min' => 1,
+      '#required' => TRUE,
+      '#default_value' => $entity->getMinimumRequiredConstraints(),
+    ];
+
+    $expire_days = $entity->getExpireDays();
+    $form['expire_password'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Expire password'),
+      '#default_value' => $expire_days > 0,
+    ];
+
+    $form['expire_days'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Expire password after'),
+      '#field_suffix' => $this->t('day(s).'),
+      '#default_value' => $expire_days ?: 1,
+      '#min' => 1,
+      '#states' => [
+        'visible' => [
+          ':input[name="expire_password"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+
+    $form['expire_warn_before_days'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Show warning'),
+      '#description' => $this->t('Show a warning message before the password would expire.<br>If set to 0 it will not show any message.'),
+      '#field_suffix' => $this->t('day(s) before the password expires.'),
+      '#default_value' => $entity->getExpireWarnDays(),
+      '#min' => 0,
+      '#states' => [
+        'visible' => [
+          ':input[name="expire_password"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+
+    $form['expiryWarningMessage'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Expiry warning message'),
+      '#description' => $this->t("This message will be presented for the user if the user's password is about to expire.<br>Useable placeholders: <ul><li>@url: URL for the user edit page</li><li>@date_time: the date and time when the password will expire</li></ul>."),
+      '#default_value' => $entity->getExpiryWarningMessage() ?? 'Your password will expire on @date_time, please <a href="@url">change your password</a> before it expires to prevent any potential data loss.',
+      '#states' => [
+        'invisible' => [
+          ':input[name="expire_warn_before_days"]' => ['value' => 0],
+        ],
+      ],
+    ];
+
+    $form['priority'] = [
+      '#type' => 'number',
+      '#title' => $this->t('Priority'),
+      '#default_value' => $entity->getPriority(),
+      '#min' => 0,
+      '#states' => [
+        'invisible' => [
+          ':input[name="role"]' => ['value' => AccountInterface::AUTHENTICATED_ROLE],
+        ],
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state): void {
+    parent::validateForm($form, $form_state);
+
+    if ($form_state->getValue('expire_warn_before_days') !== Policy::PASSWORD_NO_WARNING && empty($form_state->getValue('expiryWarningMessage'))) {
```

Comment:
Move this (and other constants) from the `Policy` class to the `PolicyInterface` interface so all implementations of this interface can inherit these constant and of the `Policy` class gets replaced by a custom module this form still can work because the new Policy class also must implement the `PolicyInterface`.